src/components/pages/blog-post/blog-post.tsx

```diff
@@ -33,114 +56,246 @@ const PanelContainer = styled.div`
   }
 `;
 
-const contentfulRenderOptions: Options = {
-  renderMark: {
-    [MARKS.CODE]: (text) => <code className="language-text">{text}</code>,
-  },
-  renderNode: {
-    [INLINES.HYPERLINK]: (props, children) =>
-      customComponents.a({
-        children,
-        href: props.data.uri,
-        target: '_blank',
-        rel: 'nofollow noopener noreferrer',
+/**
+ * Custom Contentful Rich Text Renderer
+ *
+ * Will automatically filter out ContentfulFootnote elements and appends an
+ * ordered list of footnotes after the rendered rich text.
+ *
+ * @param data rich text data
+ * @returns rendered elements
+ */
+const customContentfulRenderer = (
+  data: RenderRichTextData<ContentfulRichTextGatsbyReference>,
+) => {
+  /**
+   * Filter out footnotes from content references
+   */
+  const footnotes: ContentfulFootnoteReference[] = data.references
+    .filter(
+      (reference) =>
+        reference.__typename === ContentfulCustomModel.CONTENTFUL_FOOTNOTE,
+    )
+    .map((reference) => reference as ContentfulFootnoteReference);
+
+  /**
+   * Create an object that contains information we'll use later
+   */
+  const footnoteReferences: FootnoteReference = {};
+  footnotes.map(
+    (footnote, index) =>
+      (footnoteReferences[footnote.contentful_id] = {
+        index: index + 1,
+        anchor: `footnote-${index + 1}`,
+        referenceAnchor: `footnote-${index + 1}-reference`,
```

Comment:
So first collect all footnote references from the given data to see what we have. Then build the indexed list to have unique links to output them inside the richtext and output the actual note in the bottom.

Looks like I can follow here 🙌